aiken.html.markdown

```diff
@@ -0,0 +1,259 @@
+
+Aiken comes with a standard library and prelude that are particularly useful for creating Cardano smart contracts. The language supports various programming constructs such as functions, validators, pattern matching, and algebraic data types.
+
+Aiken accomodates to developer experience with innovations like CIP-57 (Blueprints) having been implemented. This allows developers to write less code to build transactions that are compliant with their validators.
+
+As a relatively new language, Aiken is evolving to meet the needs of the Cardano ecosystem. Its development is guided by the requirements of smart contract creation on the Cardano Blockchain.
+
+```aiken
+//// Variables & Constants
+
+// Constants are defined using 'const'
+// Constants may be used at the top level of a module
+const my_constant: Int = 42
+
+// Values assigned to let-bindings are immutable
+// New bindings can shadow previous bindings
+let x = 1
+let y = x
+let x = 2
+
+// y + x == 3
+
+//// Functions
+
+// Functions are defined using 'fn'
+
+// Functions can be named
+fn add(a: Int, b: Int) -> Int {
+  a + b
+}
+
+// Annotations are often optional (albeit recommended), as they can be inferred.
+fn add_inferred(a, b) {
+  a + b
+}
+
+// Functions are private by default. They can be exported with the 'pub' keyword.
+pub fn public_function(x: Int) -> Int {
+  x * 2
+}
+
+// Functions can accept functions as arguments
+fn apply_function_twice(f: fn(t) -> t, x) {
+  f(f(x))
+}
+
+// We can achieve the same outcome as above with pipelining using '|>'
+fn apply_function_with_pipe_twice(f: fn(t) -> t, x) {
+  x
+  |> f
+  |> f
+}
+
+// Function capturing is demonstrated here using add to create add_one
+// We choose where the argument to be passed later with '_'.
+fn capture_demonstration() {
+  let add_one = add(_, 1)
+
+  apply_function_twice(add_one, 2) // Evaluates to 4
+}
+
+// Backpassing
+
+// The simplest example provided by the documentation is as follows
+let x <- foo()
+ 
+// which is equivalent to writing:
+foo(fn(x) { })
+
+// A more in depth example follows.
+
+// We define the following set of functions over 'Fuzzer'.
+fn constant(a) -> Fuzzer<a>
+fn bool() -> Fuzzer<Bool>
+fn map(Fuzzer<a>, fn(a) -> b) -> Fuzzer<b>
+fn and_then(Fuzzer<a>, fn(a) -> Fuzzer<b>) -> Fuzzer<b>
+
+// To leverage them without backpassing is quite a syntactic mess
+pub fn option(fuzz_a: Fuzzer<a>) -> Fuzzer<Option<a>> {
+  bool()
+    |> and_then(
+         fn(predicate) {
+           if predicate {
+             fuzz_a |> map(Some)
+           } else {
+             constant(None)
+           }
+         },
+       )
+}
+
+// Whereas with backpassing the same logic may be expressed more cleanly.
+pub fn option(fuzz_a: Fuzzer<a>) -> Fuzzer<Option<a>> {
+  let predicate <- and_then(bool())
+  if predicate {
+   fuzz_a |> map(Some)
+  } else {
+    constant(None)
+  }
+}
```

Comment:
Maybe fuzzers are a little too advanced / off-topic for this. Especially since they do not introduce new syntax elements (with the exception of backpassing perhaps, but it's covered earlier already).